sound/soc/sof/core.c

```diff
@@ -362,6 +362,9 @@ int snd_sof_device_probe(struct device *dev, struct snd_sof_pdata *plat_data)
 	sdev->first_boot = true;
 	dev_set_drvdata(dev, sdev);
 
+	if (sof_core_debug)
+		dev_info(dev, "sof_debug value: %#x\n", sof_core_debug);
+
```

Comment:
I think it would make sense to print it every time and use dev_dbg instead of dev_info.

Reply:
it's true that it'll be printed every time for Intel devices since BIT(0) is always set.

Reply:
@plbossart, not in distro kernels, I think the CONFIG_SND_SOC_SOF_DEBUG_ENABLE_FIRMWARE_TRACE is not set by default.

Reply:
i am ok with dev_info.

Reply:
OK, should I re-send with some change, or?

Reply:
@ujfalusi lets merge it like that. I think the `if(sof_core_debug)` will avoid spamming production kernels.

Reply:
One thing came to mind before merging this: Do we have production kernel(s) where the NOCODEC mode is the default, which from now on can be selected via a sof_debug flag?

Reply:
I don't see how NOCODEC could be used in production kernels. That is clearly spelled out in the Kconfig

````
config SND_SOC_SOF_NOCODEC_SUPPORT
	bool "SOF nocodec static mode support"
	help
	  This adds support for a dummy/nocodec machine driver fallback
	  option if no known codec is detected. This is typically only
	  enabled for developers or devices where the sound card is
	  controlled externally.
	  This option is mutually exclusive at build time with the Intel HDAudio support.
	  Selecting it may have negative impacts and prevent e.g. microphone
	  functionality from being enabled on Intel CoffeeLake and later
	  platforms.
	  Distributions should not select this option!
````